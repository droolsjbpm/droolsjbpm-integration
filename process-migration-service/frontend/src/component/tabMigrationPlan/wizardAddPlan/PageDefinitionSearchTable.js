import React, { Component } from "react";

export default class PageDefinitionSearchTable extends Component {
  handleProcessIdChange(event) {
    this.props.handleProcessIdChange(event.target.value);
  }

  handleContainerIdChange(event) {
    this.props.handleContainerIdChange(event.target.value);
  }

  render() {
    return (
      <div>
        <div className="form-group required">
          <label className="col-sm-2 control-label">
            {this.props.tableHeader} ContainerID
          </label>
          <div className="col-sm-10">
            <input
              className="form-control"
              type="text"
              name="ContainerId"
              value={this.props.containerId}
              onChange={e => this.handleContainerIdChange(e)}
            />
          </div>
        </div>

        <div className="form-group required">
          <label className="col-sm-2 control-label">
            {this.props.tableHeader} ProcessId
          </label>
          <div className="col-sm-10">
            <input
              className="form-control"
              type="text"
              name="processId"
              value={this.props.processId}
              onChange={e => this.handleProcessIdChange(e)}
            />
          </div>
        </div>
      </div>
    );
  }
}
