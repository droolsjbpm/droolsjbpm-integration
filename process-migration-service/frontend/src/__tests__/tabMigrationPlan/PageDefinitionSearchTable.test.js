import renderer from "react-test-renderer";
import React from "react";
import PageDefinitionSearchTable from "../../component/tabMigrationPlan/wizardAddPlan/PageDefinitionSearchTable";

test("PageDefinitionSearchTable renders correctly using snapshot", () => {
  const myMock = jest.fn();
  const tree = renderer
    .create(
      <PageDefinitionSearchTable
        tableHeader="Source"
        processId="testProcessId"
        groupId="testGroupId"
        artifactId="testArtifactId"
        version="testVersion"
        handleProcessIdChange={myMock}
        handleGroupIdChange={myMock}
        handleArtifactIdChange={myMock}
        handleVersionChange={myMock}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
