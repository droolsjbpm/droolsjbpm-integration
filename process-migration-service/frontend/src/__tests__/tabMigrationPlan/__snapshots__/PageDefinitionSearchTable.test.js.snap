// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageDefinitionSearchTable renders correctly using snapshot 1`] = `
<div>
  <div
    className="form-group required"
  >
    <label
      className="col-sm-2 control-label"
    >
      Source
       Container ID
    </label>
    <div
      className="col-sm-10"
    >
      <input
        className="form-control"
        name="ContainerId"
        onChange={[Function]}
        type="text"
      />
    </div>
  </div>
  <div
    className="form-group required"
  >
    <label
      className="col-sm-2 control-label"
    >
      Source
       Process ID
    </label>
    <div
      className="col-sm-10"
    >
      <input
        className="form-control"
        name="processId"
        onChange={[Function]}
        type="text"
        value="testProcessId"
      />
    </div>
  </div>
</div>
`;
